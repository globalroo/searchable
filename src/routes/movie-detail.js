import React from "react";
import { Divider, Typography } from "@material-ui/core";

import { AssetScroller } from "src/components/asset-scroller";
import { CastMembers } from "src/components/cast-members";
import { getMovieById } from "src/tmdb-service/tmdb-api";
import { getSmallPosterImage } from "src/tmdb-service/tmdb-config";
import { useByIdLoader } from "../hooks/by-id-loader";

import "./movie-detail-style.css";

export const MovieDetail = ({ id }) => {
	const { loading, error, response } = useByIdLoader({ id, fetcher: getMovieById });
	const { payload } = response;

	//TODO: Make Loading and Error components...
	if (loading) return <div>Loading</div>;
	if (error) return <div>error</div>;

	//TODO: Break out into smaller components
	return (
		<div>
			{payload ? (
				<div className="wrapper">
					<aside className="sidebar">
						<img src={getSmallPosterImage(payload.poster_path)} alt={payload.title} />
					</aside>
					<article className="content">
						<h1 className="bold-header-uppercase">{payload.title}</h1>
						<h1>{payload.tagline}</h1>
						<div className="content-row-1">
							<Typography>{payload.overview}</Typography>
						</div>
						<Divider />
						<div className="content-row-2">
							<CastMembers cast={payload.credits.cast} />
						</div>
						<Divider />
					</article>
					{payload.recommendations.total_results > 0 && (
						<footer className="footer">
							<h1>Recommendations</h1>
							<AssetScroller assets={payload.recommendations.results} />
						</footer>
					)}
				</div>
			) : null}
		</div>
	);
};
