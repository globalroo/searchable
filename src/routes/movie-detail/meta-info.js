import React from "react";
import { Divider, Typography } from "@material-ui/core";
import { CastMembers } from "src/components/cast-members";

export const MetaInfo = ({ tagline, overview, title, cast }) => (
	<article className="content">
		<h1 className="bold-header-uppercase">{title}</h1>
		<h1>{tagline}</h1>
		<div className="content-row-1">
			<Typography>{overview}</Typography>
		</div>
		<Divider />
		<div className="content-row-2">
			<CastMembers cast={cast} />
		</div>
		<Divider />
	</article>
);
