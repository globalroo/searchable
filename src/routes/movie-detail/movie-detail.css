.content {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    overflow: hidden;
}

.content-row-1 {
    flex: 1;
    align-items: flex-start;
    justify-content: flex-start;
    flex-grow: 1
}

.content-row-2 {
    align-items: flex-end;
    justify-content: flex-end;
    overflow: hidden;
}