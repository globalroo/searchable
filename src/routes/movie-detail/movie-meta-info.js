import React from "react";
import { CastMembers } from "src/components/cast-members";
import { Divider, Typography } from "@material-ui/core";
import { format } from "date-fns";

export const MetaInfo = ({ tagline, overview, title, cast, release_date }) => (
	<article className="content">
		<h1 className="bold-header-uppercase">
			{title} [{format(release_date, "YYYY")}]
		</h1>
		<h1>{tagline}</h1>
		<div className="content-row-1">
			<Typography>{overview}</Typography>
		</div>
		<Divider />
		<div className="content-row-2">
			<CastMembers cast={cast} />
		</div>
		<Divider />
	</article>
);
