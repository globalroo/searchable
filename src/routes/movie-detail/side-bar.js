import React from "react";
import { useImageLoader } from "src/hooks/image-loader";
import { getSmallPosterImage } from "src/tmdb-service/tmdb-config";

export const SideBar = ({ poster_path, title }) => {
	const [image] = useImageLoader(getSmallPosterImage(poster_path));
	return (
		<aside className="sidebar">
			<img src={image} alt={title} />
		</aside>
	);
};
