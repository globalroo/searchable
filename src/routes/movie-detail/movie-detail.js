import React from "react";

import "./movie-detail.css";
import { AssetScroller } from "src/components/asset-scroller";
import { getMovieById } from "src/tmdb-service/tmdb-api";
import { MEDIA_TYPE } from "../../tmdb-service/tmdb-api";
import { MetaInfo } from "./movie-meta-info";
import { SideBar } from "src/components/side-bar";
import { useByIdLoader } from "src/hooks/by-id-loader";

export const MovieDetail = ({ id }) => {
	const { loading, error, response } = useByIdLoader({
		id,
		fetcher: getMovieById
	});

	const { payload } = response;

	if (loading) return null;
	if (error) return <div>Error</div>;
	if (!payload) return null;

	return (
		<div className="wrapper">
			<SideBar poster_path={payload.poster_path} title={payload.title} />
			<MetaInfo
				title={payload.title}
				release_date={payload.release_date}
				tagline={payload.tagline}
				overview={payload.overview}
				cast={payload.credits.cast}
			/>
			{payload.recommendations.total_results > 0 && (
				<footer className="footer">
					<h1>Recommendations</h1>
					<AssetScroller
						assets={payload.recommendations.results}
						link_type={MEDIA_TYPE.MOVIE}
					/>
				</footer>
			)}
		</div>
	);
};
