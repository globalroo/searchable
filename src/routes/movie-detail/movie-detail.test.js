import React from "react";
import { render } from "react-testing-library";

import { MovieDetail } from "./movie-detail";

import mockMovie from "src/test-helpers/mock-movie";

let mockError = false;
let mockLoading = false;
let mockPayload = mockMovie;

jest.mock("src/hooks/by-id-loader", () => ({
	useByIdLoader: () => ({
		response: { payload: mockPayload },
		loading: mockLoading,
		error: mockError
	})
}));

describe("Movie Detail tests", () => {
	afterEach(() => {
		mockLoading = false;
		mockError = false;
		mockPayload = mockMovie;
	});
	it("Renders a movie detail component", () => {
		const { container } = render(<MovieDetail id={mockMovie.id} />);
		expect(container).toMatchSnapshot();
	});
	it("Renders a movie detail loadiing component", () => {
		mockLoading = true;
		const { container } = render(<MovieDetail id={mockMovie.id} />);
		expect(container).toMatchSnapshot();
	});
	it("Renders a movie detail component", () => {
		mockError = true;
		const { container } = render(<MovieDetail id={mockMovie.id} />);
		expect(container).toMatchSnapshot();
	});
	it("Renders a movie detail component", () => {
		mockPayload = null;
		const { container } = render(<MovieDetail id={mockMovie.id} />);
		expect(container).toMatchSnapshot();
	});
});
