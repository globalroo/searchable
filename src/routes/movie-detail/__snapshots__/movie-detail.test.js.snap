// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Detail tests Renders a movie detail component 1`] = `
<div
  class="wrapper"
>
  <aside
    class="sidebar"
  >
    <img
      alt="Tammy"
    />
  </aside>
  <article
    class="content"
  >
    <h1
      class="bold-header-uppercase"
    >
      Tammy
       [
      2014
      ]
    </h1>
    <h1>
      She hit the road. The road hit back.
    </h1>
    <div
      class="content-row-1"
    >
      <p
        class="MuiTypography-root-1 MuiTypography-body1-10"
      >
        After losing her job and learning that her husband has been unfaithful, a woman hits the road with her profane, hard-drinking grandmother.
      </p>
    </div>
    <hr
      class="MuiDivider-root-37"
    />
    <div
      class="content-row-2"
    >
      <ul
        class="MuiGridList-root-42"
        style="margin: -2px; display: flex; flex-direction: row; flex-wrap: nowrap; transform: translateZ(0);"
      >
        <div
          style="width: 50%; height: 184px; padding: 2px; display: flex; flex-direction: column; align-items: center;"
        >
          <a
            href="/person/55536"
          >
            <div
              class="MuiAvatar-root-43 MuiAvatar-colorDefault-44"
              style="width: 80px; height: 80px; margin: 10px;"
            />
          </a>
          <p
            class="MuiTypography-root-1 MuiTypography-body1-10 MuiTypography-alignCenter-24"
          >
            Melissa McCarthy
          </p>
          <span
            class="MuiTypography-root-1 MuiTypography-caption-11 MuiTypography-alignCenter-24"
          >
            Tammy
          </span>
        </div>
      </ul>
    </div>
    <hr
      class="MuiDivider-root-37"
    />
  </article>
  <footer
    class="footer"
  >
    <h1>
      Recommendations
    </h1>
    <div
      style="display: flex; flex-wrap: wrap; justify-content: space-between; overflow: hidden;"
    >
      <ul
        class="MuiGridList-root-42"
        style="margin: -2px; flex-wrap: nowrap; transform: translateZ(0);"
      >
        <li
          class="MuiGridListTile-root-46 pointer-hover"
          style="padding: 5px;"
        >
          <div
            class="MuiGridListTile-tile-47"
          >
            <img
              alt="The Heat"
              style="height: 225px;"
            />
            <div
              class="MuiGridListTileBar-root-50 MuiGridListTileBar-titlePositionBottom-51"
            >
              <div
                class="MuiGridListTileBar-titleWrap-54 MuiGridListTileBar-titleWrapActionPosRight-56"
              >
                <div
                  class="MuiGridListTileBar-title-57"
                >
                  The Heat
                </div>
              </div>
              <div
                class="MuiGridListTileBar-actionIcon-59"
              >
                <button
                  class="MuiButtonBase-root-67 MuiIconButton-root-61"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label-66"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root-70"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root-79"
                  />
                </button>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </footer>
</div>
`;
